pkg/controller/installconfig/installconfig_controller.go

```diff
@@ -0,0 +1,211 @@
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a InstallConfig object and makes changes based on the state read
+// and what is in the InstallConfig.Spec
+// Note: The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileInstallConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling InstallConfig")
+
+	// Fetch the InstallConfig instance
+	installConfig := &v1alpha1.InstallConfig{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, installConfig)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	if err := r.EnsureCheSubscription(reqLogger, installConfig); err != nil {
+		reqLogger.Error(err, "failed to create subscription for che", "Che.Namespace", installConfig.Spec.CheOperatorSpec.Namespace)
+		return r.StatusUpdate(reqLogger, installConfig, r.setStatusCheSubscriptionFailed, err.Error())
+	}
+
+	return r.StatusUpdate(reqLogger, installConfig, r.setStatusCheSubscriptionCreated, "che operator subscription created")
+}
+
+func (r *ReconcileInstallConfig) EnsureCheSubscription(logger logr.Logger, installConfig *v1alpha1.InstallConfig) error {
+	ns, err := r.ensureCheNamespace(logger, installConfig)
+	if err != nil {
+		return err
+	}
+
+	if err := r.createCheOperatorGroup(logger, ns, installConfig); err != nil {
+		return err
+	}
+
+	return r.createCheSubscription(logger, ns, installConfig)
+}
+
+func (r *ReconcileInstallConfig) ensureCheNamespace(logger logr.Logger, installConfig *v1alpha1.InstallConfig) (string, error) {
+	cheOpNamespace := installConfig.Spec.CheOperatorSpec.Namespace
+	ns := &v1.Namespace{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: cheOpNamespace}, ns)
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Creating a namespace for che operator", "Namespace", cheOpNamespace)
+		namespace := che.Namespace(cheOpNamespace)
+		if err := controllerutil.SetControllerReference(installConfig, namespace, r.scheme); err != nil {
+			return cheOpNamespace, err
+		}
+		return cheOpNamespace, r.client.Create(context.TODO(), namespace)
+	}
+
+	return cheOpNamespace, err
+}
+
+func (r *ReconcileInstallConfig) createCheOperatorGroup(logger logr.Logger, ns string, installConfig *v1alpha1.InstallConfig) error {
+	operatorGroup := che.OperatorGroup(ns)
```

Comment:
Maybe
```suggestion
	operatorGroup := che.NewOperatorGroup(ns)
```
would be more clear? WDYT?

Reply:
done in  27ea567